deeppavlov/models/doc_retrieval/logit_ranker.py

```diff
@@ -33,50 +32,69 @@ class LogitRanker(Component):
-
+             a batch of best answers, their scores, places in contexts
+             and tfidf_doc_ids for this answers if tfidf_doc_ids were passed
         """
-        # TODO output result for top_n
-        warnings.warn(f'{self.__class__.__name__}.__call__() API will be changed in the future release.'
-                      ' Instead of returning Tuple(List[str], List[float] will return'
-                      ' Tuple(List[List[str]], List[List[float]]).', FutureWarning)
+        if tfidf_doc_ids is None:
+            logger.info("you didn't pass tfidf_doc_ids as input in logit_ranker config so batch_best_answers_doc_ids"
+                        "can't be compute")
 
         batch_best_answers = []
-        batch_best_answers_scores = []
-        for contexts, questions in zip(contexts_batch, questions_batch):
+        batch_best_answers_score = []
+        batch_best_answers_place = []
+        batch_best_answers_doc_ids = []
+        for quest_ind, [contexts, questions] in enumerate(zip(contexts_batch, questions_batch)):
             results = []
             for i in range(0, len(contexts), self.batch_size):
                 c_batch = contexts[i: i + self.batch_size]
                 q_batch = questions[i: i + self.batch_size]
                 batch_predict = zip(*self.squad_model(c_batch, q_batch))
                 results += batch_predict
             if self.sort_noans:
-                results = sorted(results, key=lambda x: (x[0] != '', x[2]), reverse=True)
+                results_sort = sorted(results, key=lambda x: (x[0] != '', x[2]), reverse=True)
             else:
-                results = sorted(results, key=itemgetter(2), reverse=True)
-            batch_best_answers.append(results[0][0])
-            batch_best_answers_scores.append(results[0][2])
-        return batch_best_answers, batch_best_answers_scores
+                results_sort = sorted(results, key=itemgetter(2), reverse=True)
+            batch_best_answers.append([x[0] for x in results_sort[:self.top_n] if x[0]])
+            batch_best_answers_place.append([x[1] for x in results_sort[:self.top_n] if x[0]])
+            batch_best_answers_score.append([x[2] for x in results_sort[:self.top_n] if x[0]])
+            if tfidf_doc_ids is not None:
+                doc_ind = [results.index(x) for x in results_sort]
+                batch_best_answers_doc_ids.append(
+                    [tfidf_doc_ids[quest_ind][i] for i in doc_ind][:len(batch_best_answers[-1])])
+
+        if self.top_n == 1:
+            batch_best_answers = [x[0] for x in batch_best_answers]
+            batch_best_answers_place = [x[0] for x in batch_best_answers_place]
+            batch_best_answers_score = [x[0] for x in batch_best_answers_score]
+            batch_best_answers_doc_ids = [x[0] for x in batch_best_answers_doc_ids]
+
+        return batch_best_answers, batch_best_answers_score, batch_best_answers_place, batch_best_answers_doc_ids
```

Comment:
return 3 values in case if tfidf_doc_ids is None
```suggestion
        if tfidf_doc_ids is None:
            return batch_best_answers, batch_best_answers_score, batch_best_answers_place

        return batch_best_answers, batch_best_answers_score, batch_best_answers_place, batch_best_answers_doc_ids
```